src/acrcssc/azext_acrcssc/helper/_taskoperations.py

```diff
@@ -99,10 +103,37 @@ def _create_cssc_workflow(cmd, registry, schedule_cron_expression, resource_grou
         dry_run
     )
 
-    logger.warning(f"Deployment of {CONTINUOUS_PATCHING_WORKFLOW_NAME} tasks completed successfully.")
+    if not silent_execution:
+        print(f"Deployment of {CONTINUOUS_PATCHING_WORKFLOW_NAME} tasks completed successfully.")
+
 
+def _update_cssc_workflow(cmd, registry, schedule_cron_expression, resource_group, dry_run, task_list):
+    # compare the task definition to the existing tasks, if there is a difference, we need to update the tasks
+    # if we need to update the tasks, we will update the cron expression from it
+    # if not we just update the cron expression from the given parameter
+    for task in task_list:
+        if task.name not in CONTINUOUSPATCH_ALL_TASK_NAMES:
+            logger.debug(f"Task {task.name} is not part of the continuous patching workflow, skipping update")
+            continue
+        deployed_task = task.step.encoded_task_content
+        extension_task = _create_encoded_task(CONTINUOUSPATCH_TASK_DEFINITION[task.name]["template_file"])
+        if deployed_task != extension_task:
+            logger.debug(f"Task {task.name} is different from the extension task, updating the task")
+
+            # TODO this is wrong, we don't know if the current taks is the trigger with the schedule
```

Comment:
I don't quite understand the code logic here. Since the task name is hardcoded, we should be able to determine if the task is a triggered one by its name. If the goal is to update the schedule, if schedule_cron_expression is None, it means we don't need to pass any value, allowing the patch task to retain its original schedule.

Reply:
The above `TODO` comment is outdated and no longer applies.
The logic here is to reuse the existing ARM template to update the tasks, and the cron expression is part of the parameters of the template. If we send an empty parameter, we will be patching over with a null cron expression.